test/integration/end_to_end.exs

```diff
@@ -0,0 +1,70 @@
+defmodule Mississippi.EndToEnd.Test do
+  use ExUnit.Case
+
+  import Hammox
+
+  alias Mississippi.Consumer.AMQPDataConsumer.ExRabbitPoolConnection
+  alias Mississippi.Producer.EventsProducer
+
+  require Logger
+
+  @moduletag :integration
+
+  setup_all do
+    stub_with(MockAMQPConnection, ExRabbitPoolConnection)
+    Hammox.set_mox_global()
+    queue_count = System.unique_integer([:positive])
+
+    prefix = "mississippi_test_#{System.unique_integer()}_"
+    # We use the default exchange so that queues are binded using the routing key
+    exchange_name = ""
+
+    producer_options = [
+      amqp_producer_options: [host: "localhost"],
+      mississippi_config: [
+        queues: [events_exchange_name: exchange_name, total_count: queue_count, prefix: prefix]
+      ]
+    ]
+
+    consumer_options = [
+      amqp_consumer_options: [host: "localhost"],
+      mississippi_config: [
+        queues: [
+          events_exchange_name: exchange_name,
+          prefix: prefix,
+          range_start: 0,
+          range_end: queue_count - 1,
+          total_count: queue_count
+        ],
+        message_handler: E2EMessageHandler
+      ]
+    ]
+
+    %{
+      producer: start_supervised!({Mississippi.Producer, producer_options}),
+      consumer: start_supervised!({Mississippi.Consumer, consumer_options})
+    }
+  end
+
+  setup do
+    E2EMessageHandler.start_with_receiver(self())
+
+    %{
+      sharding_key: "sharding_key_#{System.unique_integer()}",
+      payload: "payload_#{System.unique_integer()}",
+      timestamp: DateTime.to_unix(DateTime.utc_now())
+    }
+  end
+
+  @tag :e2e
+  test "Message is published and received", %{
```

Comment:
can we also check the opposite behavior? error codes?

Reply:
I would not expect an e2e test that fails, however there might be some more testing done on the producer side. @noaccOS' #11 is a step in that direction. In my opinion, another PR can be added to unit-test publishing errors (e.g. missing sharding key), without need to interact with RabbitMQ